File: examples/payload_get_fov_status.cpp
/**
 * This sample will show you how to get the FOV_STATUS from the payload
 **/

#include "stdio.h"
#include"payloadSdkInterface.h"

#include <iostream>
#include <chrono>

PayloadSdkInterface* my_payload = nullptr;

#if (CONTROL_METHOD == CONTROL_UART)
T_ConnInfo s_conn = {
    CONTROL_UART,
    payload_uart_port,
    payload_uart_baud
};
#else
T_ConnInfo s_conn = {
    CONTROL_UDP,
    udp_ip_target,
    udp_port_target
};
#endif

void quit_handler(int sig);
void onPayloadStatusChanged(int event, double* param);

int main(int argc, char *argv[]){
	printf("Starting SendSystemTime example...\n");
	signal(SIGINT,quit_handler);

	// creat payloadsdk object
	my_payload = new PayloadSdkInterface(s_conn);

	// register callback function
	my_payload->regPayloadStatusChanged(onPayloadStatusChanged);

	// init payload
	my_payload->sdkInitConnection();
	printf("Waiting for payload signal! \n");

	// check connection
	my_payload->checkPayloadConnection();

	int msg_cnt = 0;
	int boot_time_ms = 0;

	while(1){
		
		my_payload->getPayloadCameraFOVStatus(CAMERA_EO);
		usleep(500000); // 500ms, 2Hz

		my_payload->getPayloadCameraFOVStatus(CAMERA_IR);
		usleep(500000); // 500ms, 2Hz
	}

	return 0;
}

void quit_handler( int sig ){
    printf("\n");
    printf("TERMINATING AT USER REQUEST \n");
    printf("\n");

    // close payload interface
    try {
        my_payload->sdkQuit();
    }
    catch (int error){}

    // end program here
    exit(0);
}

void onPayloadStatusChanged(int event, double* param){
	// return;
	switch(event){
	case PAYLOAD_PARAM_CAM_FOV_STATUS:{
		// param[0]: camera id
		// param[1]: hfov
		// param[2]: vfov

		printf("Camera ID: %.2f - HFOV: %.2f - VFOV: %.2f \n", param[0], param[1], param[2]);
		break;
	}
	default: break;
	}
}

File: examples/payload_get_fov_status.py
#!/usr/bin/env python3
import sys
import os

# Add the libs directory to the path
sys.path.append(os.path.join(os.path.dirname(__file__), '..', 'libs'))

# Import config first to setup environment automatically
from config import config

import time
import signal
import sys
from payload_sdk import PayloadSdkInterface, payload_status_event_t, camera_type_t
from payload_define import *

my_payload = None

# Signal handler for quitting
def quit_handler(sig, frame):
    global my_payload
    print("\nTERMINATING AT USER REQUEST")

    # Close payload interface
    if my_payload:
        try:
            my_payload.sdkQuit()
        except Exception as e:
            print(f"Error while quitting payload: {e}")
    # End program 
    sys.exit(0)

# Callback function for payload status changes
def onPayloadStatusChanged(event: int, param: list):
    if payload_status_event_t(event) == payload_status_event_t.PAYLOAD_PARAM_CAM_FOV_STATUS:
        # param[0]: camera id
        # param[1]: hfov
        # param[2]: vfov
        print(f"HFOV: {param[1]:.2f} - VFOV: {param[2]:.2f}")

def main():
    global my_payload

    print("Starting Set gimbal mode example...")
    signal.signal(signal.SIGINT, quit_handler)

    # Create payloadsdk object
    my_payload = PayloadSdkInterface()

    # Register callback function
    my_payload.regPayloadStatusChanged(onPayloadStatusChanged)

    # Init payload
    my_payload.sdkInitConnection()
    print("Waiting for payload signal!")

    # Check connection
    my_payload.checkPayloadConnection()
    
    while True:
        my_payload.getPayloadCameraFOVStatus(camera_type_t.CAMERA_EO)
        time.sleep(0.5)  # 500ms, 2Hz

        my_payload.getPayloadCameraFOVStatus(camera_type_t.CAMERA_IR)
        time.sleep(0.5)  # 500ms, 2Hz

if __name__ == "__main__":
    main()